t/node/healthcheck-metric.t

```diff
@@ -0,0 +1,491 @@
+                    ngx.say("failed to request: ", err)
+                    return
+                end
+                local apisix_upstream_status_body = find_apisix_upstream_status(metric_res.body)
+                for _, match in ipairs(apisix_upstream_status_body) do
+                    ngx.say(match)
+                end
+            end
+
+            -- create a route
+            local data = {
+                uri = "/ping",
+                upstream = {
+                    nodes = {
+                        ["127.0.0.1:8765"] = 1,
+                        ["127.0.0.1:8766"] = 1,
+                        ["127.0.0.1:8767"] = 1
+                    },
+                    retries = 2,
+                    checks = {
+                        active = {
+                            http_path = "/healthz",
+                            healthy = {
+                                interval = 1,
+                                successes = 1
+                            },
+                            unhealthy = {
+                                interval = 1,
+                                http_failures = 1
+                            },
+                        }
+                    }
+                }
+            }
+            local code, body = t.test('/apisix/admin/routes/1',
+                ngx.HTTP_PUT, core.json.encode(data))
+            if code >= 300 then
+                ngx.status = code
+                ngx.say(body)
+                return
+            end
+
+            local httpc = http.new()
+            local uri = "http://127.0.0.1:" .. ngx.var.server_port .. "/ping"
+            local _, _ = httpc:request_uri(uri, {method = "GET", keepalive = false})
+            ngx.sleep(3)
+
+            --- get metrics
+            get_metrics()
+
+            ngx.say("update the upstream")
+
+            --- update the route
+            local new_data = {
+                uri = "/ping",
+                upstream = {
+                    nodes = {
+                        ["127.0.0.1:8765"] = 1,
+                        ["127.0.0.1:8766"] = 1
+                    },
+                    retries = 2,
+                    checks = {
+                        active = {
+                            http_path = "/healthz",
+                            healthy = {
+                                interval = 1,
+                                successes = 1
+                            },
+                            unhealthy = {
+                                interval = 1,
+                                http_failures = 1
+                            },
+                        }
+                    }
+                }
+            }
+            local code, body = t.test('/apisix/admin/routes/1',
+                ngx.HTTP_PUT, core.json.encode(new_data))
+            if code >= 300 then
+                ngx.status = code
+                ngx.say(body)
+                return
+            end
+            ngx.sleep(6)
```

Comment:
It takes too long, try to replace `delayed_clear` with `clear` when run test cases

Reply:
This time is set after many attempts. 3, 4, and 5 have a small probability of causing the target to not be cleaned up in time enough.

I don't agree with using clear in test cases, it will make the test cases complicated and difficult to read.
1. The checker cannot be obtained directly in the test case.
2. The underlying health check library does not support directly calling functions to delete targets.